core/trino-main/src/main/java/io/trino/operator/join/JoinUtils.java

```diff
@@ -67,7 +67,7 @@ public static boolean isBuildSideReplicated(PlanNode node)
                             planNode -> planNode instanceof ProjectNode ||
                                     isLocalRepartitionExchange(planNode) ||
                                     isLocalGatherExchange(planNode))  // used in cross join case
-                    .where(joinNode -> isRemoteReplicatedExchange(joinNode) || isRemoteReplicatedSourceNode(joinNode))
+                    .where(rightNode -> isRemoteReplicatedExchange(rightNode) || isRemoteReplicatedSourceNode(rightNode))
```

Comment:
Or it could be `planNode` since it is more of a predicate.

Reply:
Search recursively from `JoinNode.rightNode`(call it rightNode):
- If the child exist `ProjectNode ` or `isLocalRepartitionExchange` or `isLocalGatherExchange`, then execute `where` logic.
- When `where` logic be executed, `where` argument should still be `rightNode`

so I think rightNode is more reasonable, FYI.

Reply:
i agree this should be `planNode`

(it can be any child of the join's right node)

Reply:
I'm ok with `planNode `, updated